application/src/main/java/org/togetherjava/tjbot/commands/help/AskCommand.java

```diff
@@ -146,6 +132,40 @@ private boolean isUserOnCooldown(User user) {
             .isPresent();
     }
 
+    private void sendCooldownResponse(SlashCommandInteractionEvent event) {
+        User user = event.getUser();
+
+        String message =
+                """
+                        Sorry, you can only create a single help thread every 5 minutes. Please use your existing thread %s instead.
+                        If you made a typo or similar, you can adjust the title using the command %s and the category with %s :ok_hand:""";
+
+        RestAction<String> changeTitle = MessageUtils.mentionGuildSlashCommand(event.getGuild(),
+                HelpThreadCommand.COMMAND_NAME, HelpThreadCommand.CHANGE_SUBCOMMAND_GROUP,
+                HelpThreadCommand.CHANGE_TITLE_SUBCOMMAND);
+        RestAction<String> changeCategory = MessageUtils.mentionGuildSlashCommand(event.getGuild(),
+                HelpThreadCommand.COMMAND_NAME, HelpThreadCommand.CHANGE_SUBCOMMAND_GROUP,
+                HelpThreadCommand.CHANGE_CATEGORY_OPTION);
+        long lastCreatedThreadId = database
+            .read(context -> context.selectFrom(HELP_THREADS)
+                .where(HELP_THREADS.AUTHOR_ID.eq(user.getIdLong()))
+                .orderBy(HELP_THREADS.CREATED_AT.desc())
+                .fetch())
+            .get(0)
+            .getChannelId();
+
+        if (lastCreatedThreadId == 0) {
+            logger.warn("Can't find the last help thread created by the user with id ({})",
+                    user.getId());
+        }
```

Comment:
this is incorrect. feels like u didnt test it.

if there is no thread, it will fail on your `get(0)` already, not down here.

u have to test code, people trust u on this...

Reply:
@Zabuzard will it still be the case if i do .fetchOne and not get(0) is there a way to get 0 by default?

Reply:
fetchOne returns an `Optional`

Reply:
or null or sth. point is, it tells u when it didnt find anything and u should check on that. like
```java
Foo lastThreadByAuthor = database...

if (lastThreadByAuthor == null) {
  ...
  return;
}

String lastThreadByAuthorMention = MessageUtils.mentionChannelById(lastThreadByAuthor.getChannelId();
...
```

Reply:
primitives can't be null no? it'll get to the default value which is 0 iirc

Reply:
but it doesnt return a primitive. it returns the record. and u call a method on the record. but its null and crashes.

please for god sake, **test** your code. it is mandatory that you do that. trigger all edge cases at least once to verify that they work. its so easy to make mistakes and u cant always rely on people seeing it in code reviews. we will miss something.

Reply:
still wrong.

Reply:
done